scripts/style-spec.js

```diff
@@ -1 +1,168 @@
+        },
+        "doc": "Name of image in sprite to use as the background of the location component."
+    },
+    "location": {
+        "type": "array",
+        "default": [
+            0.0,
+            0.0,
+            0.0
+        ],
+        "length": 3,
+        "value": "number",
+        "property-type": "data-constant",
+        "expression": {
+            "interpolated": false,
+            "parameters": []
+        },
+        "doc": "An array of [latitude, longitude, altitude] position of the location component."
+    },
+    "bearing": {
+        "type": "number",
+        "default": "0",
+        "property-type": "data-constant",
+        "expression": {
+            "interpolated": false,
+            "parameters": [ ]
+        },
+        "doc": "The bearing of the location component."
+    },
+    "accuracy-radius": {
+        "type": "number",
+        "default": "0",
+        "property-type": "data-constant",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
+        "doc": "The accuracy, in meters, of the position source used to retrieve the position of the location component."
+    },
+    "top-image-size": {
+        "type": "number",
+        "property-type": "data-constant",
+        "default": "1",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
+        "doc": "The size of the top image, in pixels."
+    },
+    "bearing-image-size": {
+        "type": "number",
+        "property-type": "data-constant",
+        "default": "1",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
+        "doc": "The size of the bearing image, in pixels."
+    },
+    "shadow-image-size": {
+        "type": "number",
+        "property-type": "data-constant",
+        "default": "1",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
+        "doc": "The size of the shadow image, in pixels."
+    },
+    "perspective-compensation": {
+        "type": "number",
+        "default": "0.85",
+        "property-type": "data-constant",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
+        "doc": "The amount of the perspective compensation, between 0 and 1. A value of 1 produces a location component of constant width across the screen. A value of 0 makes it scale naturally according to the viewing projection."
+    },
+    "image-tilt-displacement": {
+        "type": "number",
+        "property-type": "data-constant",
+        "default": "0",
+        "expression": {
+            "interpolated": true,
+            "parameters": [
+                "zoom"
+            ]
+        },
```

Comment:
They are effectively used in layouting (like, for example, the size of the radius, or the shift of the 3d effect). I believe we can change them into paint properties if we desire to enable transitions on them (perhaps the sizes of the rasters?). It would still trigger relayouting, that in this case is perhaps relatively cheap.

Reply:
This layer is neither backed by tiles, nor require layout. All properties could be transitionable.

```
LayerTypeInfo::Layout::NotRequired,
LayerTypeInfo::FadingTiles::NotRequired,
LayerTypeInfo::CrossTileIndex::NotRequired,
LayerTypeInfo::TileKind::NotRequired
```

Reply:
yes, here by layout i meant recomputing the geometry, as opposed to simply pass in values that will be set as shader uniforms.